### PR TITLE
feat(api): Added configurable min and max volume to pipettes

### LINKED ISSUE #2075: Feature request: Ability to set min_volume of pipettes
## overview

A feature to be able to set the minimum volume of a pipette by keyword argument.

There are situations when I would like to select which pipette to use depending on the volume being transferred, and comparing the transfer volume against the pipette's minimum volume instead of a magic number seems like the best way to do that.

## current behavior

The minimum volume is set to 0 by default for the new pipette classes, `P10_Single` and `P300_Single`. Currently the only way to set the `min_volume` is by patching the pipette instance after it is created:

    p300 = P300_Single(mount='left', tip_racks=[p300_tips])
    p300.min_volume = 30

    p10 = P10_Single(mount ='right', tip_racks=[p10_tips])
    p10.min_volume = 1

The attribute cannot be set by keyword arg as it could be in the now deprecated `Pipette` class. This could also lead to uncaught bugs caused by typos, e.g:

    p10.min_vol = 1

## expected behavior

Ability to set `min_volume` as a keyword argument in new pipette constructors. The default value should be a sensible 10% of the `max_volume` as described in #1603. If #1603 is implemented, the nice alert banner will never see the light of day for most users unless they explicitly set the value in every script.

This feels a lot cleaner than the current method:

    p300 = P300_Single(mount='left', tip_racks=[p300_tips], min_volume=20)
    p10 = p10_Single(mount='right', tip_racks=[p10_tips])

### PR DESCRIPTION
**Pull request by rvinzent**
## overview

As a user, I would like to configure the pipette min and max volumes in the constructor of the class. Before, the only way to set the `min_volume` and `max_volume` of a pipette was to patch the pipette instance after it is constructed. This feels hacky and is prone to errors. After this change, you can set pipette min and max volume in the constructor of the class to set these attributes.

Default values of 10% of the max volume were also added in the pipette config.

Closes #2075 

## changelog

- `min_volume` and `max_volume` keywords added to pipette constructors
- `"minVolume"` field added to `pipette-config.json`

### CODE REVIEW

**Comment by btmorr** [comment]
Hi @rvinzent ! Thanks for the contribution. I've checked out your fork on my laptop and I'm checking the tests to see why they're failing (it's possible that the tests are written in a way that is brittle). I'll let you know as soon as I've figured that out so we can get this in a mergable state.

**Comment by rvinzent** [comment]
I didn't look too much into it, but if I recall correctly it looked like there were more notifications than expected during the `dinosaur.py` test. Since this PR changes the default `min_volume`, the default `disposal_vol` for a `distrubute` command is some value rather than `0`. This may result in some liquid being left in the pipette tip when it goes to the trash, as expected.

It looks like the robot will blow out any liquid remaining in the tip before dropping it in the trash as a result of this fucntion in `instruments.py`:

    def _blowout_during_transfer(self, loc, **kwargs):
        blow_out = kwargs.get('blow_out', False)
    
        # Looks like it decides to blow out over the trash here
        if self.current_volume > 0 or blow_out:
            if not isinstance(blow_out, Placeable):
                blow_out = self.trash_container
                if self.current_volume == 0:
                    blow_out = None
            self.blow_out(blow_out)
            self._mix_during_transfer(
                kwargs.get('mix_after', (0, 0)),
                loc,
                **kwargs)

I believe this would result in an additional notification for blowing out the liquid where it wasn't previously happening because `disposal_vol` was zero. I would argue that the blow out over the trash container isn't necessary and we can just drop the tip along with any remaining liquid into the trash.

For me, the failures are probably a result of expected behavior, but changing the `_blow_out_during_transfer` method to ignore remaining liquid in the tip before dropping it might break other tests, and might further change the number of actions that are counted in the two failing tests. If this is indeed the reason for the failures then perhaps we could simply update the number of actions in the tests to reflect the new behavior resulting from a non-zero default `disposal_vol`.

Hope this helps!

**Comment by btmorr** [comment]
Ok, I've finally been able to track down the issue that is causing tests to fail. The tests on distribute have been written in a way that actually enshrine a misbehavior, and this PR fixes a bug that we were not aware of because of that test.

The "dinosaur" protocol that is used for testing uses the `distribute` method with a P300 Single, distributing 50ul to a set of wells, and if `min_volume` is 0, then this gets broken down into an aspirate of 300ul and 6 dispenses of 50ul each (the behavior that is asserted against in the test). The default behavior is that each `aspirate` should be paired with a `blow_out` of the minimum volume, which is now non-zero, thus changing the sequence to be an aspirate of 280ul (5*50 + 30) and then 5 dispenses of 50ul each and a blowout of 30ul into the trash. 

In short, this PR causes the `distribute` command to behave as intended. In order to get it merged, you can do one of two things:
1. You can modify the two affected tests ("test_server.py::test_notifications" and "test_session.py:: test_load_from_text") to match the desired sequence.
2. You can add `disposal_vol=0` to the two dispense commands in "api/tests/opentrons/data/dinosaur.py", and then in a follow-up PR one of the engineers at Opentrons will improve the tests.

In either case, one of us will work on improving the documentation around this behavior. Let me know if you need any further assistance.

**Comment by rvinzent** [comment]
Hi Ben, I updated the disposal volume in `dinosaur.py` and merged with the latest edge branch and the build is passing. I'm not familiar with squashing commits though and I would appreciate a bit of an explanation on how I would go about doing that.

Thanks!
